src/procgen/Biome.ts

```diff
@@ -196,14 +211,41 @@ export class Biome {
 
   preprocessed = new Map<BiomeLandKey, PreprocessedLandConf>()
 
-  #initialized = false
+  constructor() {
+    this.heatmap = new ProcLayer('heatmap')
+    this.heatmap.sampling.harmonicsCount = 6
+    this.heatmap.sampling.periodicity =
+      worldRootEnv.rawSettings.biomes.periodicity
+    this.rainmap = new ProcLayer('rainmap')
+    this.rainmap.sampling.harmonicsCount = 6
+    this.rainmap.sampling.periodicity =
+      worldRootEnv.rawSettings.biomes.periodicity
+    // const mappingProfile = MappingProfiles[ProfilePreset.Stairs2]()
+    // this.heatProfile = LinkedList.fromArrayAfterSorting(mappingProfile, MappingRangeSorter)  // 3 levels (COLD, TEMPERATE, HOT)
+    // this.rainProfile = LinkedList.fromArrayAfterSorting(mappingProfile, MappingRangeSorter) // 3 levels (DRY, MODERATE, WET)
+    this.posRandomizer = new ProcLayer('pos_random')
+    this.posRandomizer.sampling.periodicity = 6
+    const isEmptyConf = Object.keys(Biome.externalRawConf).length === 0
+    if (!isEmptyConf) {
+      this.parseBiomesConfig(Biome.externalRawConf)
+    } else {
+      console.warn(
+        `missing biome configuration at creation, must provide conf later`,
+      )
```

Comment:
A warning is not useful and only for human consumption, we want to strictly fail if there is no configuration